rpc/provider/smoldot.ts

```diff
@@ -67,27 +92,24 @@ async function connection(
   }
   let conn = connections.get(chainSpec);
   if (!conn) {
-    // TODO: try catch this and send through handler within a `ProviderHandlerError`
     const inner = await client.addChain({ chainSpec });
```

Comment:
for parachain support, all you need to do is to pass in the potentialRelayChains all the chains inner that you collect during the addChain. Meaning something like:
```suggestion
    const innerMap: Chain[] = [];
    connections.forEach((value: SmoldotProviderConnection) => {
      innerMap.push(value.inner);
    });
    const inner = await client.addChain({ chainSpec, potentialRelayChains: innerMap });
```

Reply:
thx @wirednkod 
The parachain support will be done in a separate PR

Reply:
@harrysolovay we need to decide 2 things

- 1st when connecting to a parachain, I believe that we need to pass the parachain spec and the potential relay chain spec
- 2nd smoldot connection clean up, given that it takes time to sync the light client, connecting/disconnecting to the same chain will make it slow, so perhaps we can keep the synced state without closing the connections

Thouths?

Reply:
> thx @wirednkod The parachain support will be done in a separate PR

#378